src/asteroid_tracker/asteroid_tracker.py

```diff
@@ -13,50 +16,122 @@
+
+    # end_date = end_date.strip()
+
+    stats = calculate_statistics(start_date, end_date)
     print_statistics(stats)
 
 
 def calculate_statistics(start_date, end_date):
     """
     Make an API request and calculate statistics.
     """
-
     # TODO: Task 2 - Prepare and make the HTTP request
+    global format_Error
+    try:
+        datetime.datetime.strptime(start_date, '%Y-%m-%d')
+    except ValueError:
+        format_Error = True
 
+    try:
+        datetime.datetime.strptime(end_date, '%Y-%m-%d')
+    except ValueError:
+        format_Error = True
 
-    num_asteroids = 0
-    num_potentially_hazardous_asteroids = 0
+    d_error = {'code': 400, 'type': 'BAD_REQUEST', 'message': 'Date Format Exception - Expected format (yyyy-mm-dd) - The Feed date limit is only 7 Days', }
+    dict_error = {'error': d_error}
 
-    largest_diameter_meters = -1
-    nearest_miss_kms = -1
+    if format_Error:
+        return dict_error
 
-    # TODO: Task 3 - Calculate statistics
-
-    return {
-        'start_date': start_date,
-        'end_date': end_date,
-        'num_asteroids': num_asteroids,
-        'num_potentially_hazardous_asteroids': num_potentially_hazardous_asteroids,
-        'largest_diameter_meters': largest_diameter_meters,
-        'nearest_miss_kms': nearest_miss_kms,
-    }
+    else:
+        url = API_URL % (start_date, end_date)
+        response = requests.get(url)
+        # print(response.content)
+
+        json_data = json.loads(response.content)
+        # print(json.dumps(json_data, indent = 4))
+
+        # num_asteroids
+        keyVal = 'element_count'
+        if keyVal in json_data:
+            num_asteroids = json_data[keyVal]
+
+        # num_potentially_hazardous_asteroids
+        num_potentially_hazardous_asteroids = 0
+        count = 0
+
+        for nearEarthObject in json_data['near_earth_objects']:
+            for item in json_data['near_earth_objects'][nearEarthObject]:
+                if item['is_potentially_hazardous_asteroid']:
+                    count = count + 1
+        num_potentially_hazardous_asteroids = count
+
+        # largest_estimated_diameter_meters
+        maximum = 0
+        min_diam = 0
+
+        # print(json_data['near_earth_objects'][end_date][0][keyVal]['meters']['estimated_diameter_max'])
+        for nearEarthObject in json_data['near_earth_objects']:
+            for item in json_data['near_earth_objects'][nearEarthObject]:
+                val = item['estimated_diameter']['meters']['estimated_diameter_max']
+                if val > maximum:
+                    maximum = val
+                    min_diam = item['estimated_diameter']['meters']['estimated_diameter_min']
+        largest_diameter_meters = (maximum + min_diam) / 2
+
+        # nearest_miss_kms
+        nearest_miss_kms = -1
+        minimum = 0
+        minimum = json_data['near_earth_objects'][end_date][0]['close_approach_data'][0]['miss_distance']['kilometers']
```

Comment:
This line looks very dodgy. You shouldn't ever have to write lines like this. For scenarios like this we usually use an initialization value like `-1` or `None` that we know will never occur in the data. Then we update the `if` statement below such that we select the first value we encounter as the minimum during the loop:
```python
if minimum == -1 or float(item2['miss_distance']['kilometers']) < float(minimum):
```